REQUIREMENTS.md

```diff
@@ -0,0 +1,155 @@
+# IPLD Requirements
+
+This document outlines parts of IPLD that should and should not be changed to
+ensure the success of future improvements (especially type systems).
+
+**Definitions**
+
+* Block: A block is a chunk of an IPLD DAG, encoded in a format. Blocks have CIDs.
+* Patch: A piece of an IPLD DAG. Blocks contain patches.
+* Node: A node is a *point* in an IPLD DAG (object, array, number, etc.).
+* Link: A link is an IPLD Node that points to another IPLD Node.
+* Path: A paths a human readable pointer to an IPLD Node.
```

Comment:
I like the distinction this is starting to draw between blocks and nodes/links/paths. These lines have been pretty blurry in prior docs and I like this way of drawing the distinction.

Reply:
This came out of a discussion with @jbenet in London (which never made it to GitHub).